waspc/examples/todoApp/src/client/pages/ProfilePage.tsx

```diff
@@ -16,14 +16,14 @@ export const ProfilePage = ({
   user: User
 }) => {
   const [messages, setMessages] = useState<{ id: string, username: string, text: string }[]>([])
-  const { socket, isConnected, registerHandler } = useSocket<ServerToClientEvents, ClientToServerEvents>()
+  const { socket, isConnected } = useSocket<ServerToClientEvents, ClientToServerEvents>()
   const inputRef = useRef<HTMLInputElement>(null)
 
   useEffect(() => {
     fetchCustomRoute()
   }, [])
 
-  registerHandler('chatMessage', logMessage)
+  useSocketListener(socket, 'chatMessage', logMessage)
```

Comment:
Does it have to receive socket? Socket is global and known, so it could just obtain it on its own?

Reply:
This is what we use to infer the type. It either has to take a socket (as suggested in the Discord chat) or we have to accept the type (like I had before). Pulling off socket feels cleaner to me.

Reply:
Ok, that does make sense. Also it is kind of clear from the code the connection between the two.